src/commands/build.ts

```diff
@@ -20,42 +21,48 @@ import {
 } from './build-util';
 import {stopEsbuild} from './esbuildPlugin';
 import {createImportResolver} from './import-resolver';
-import {command as installCommand} from './install';
+import {getInstallTargets, run as installRunner} from './install';
 import {paint} from './paint';
 import srcFileExtensionMapping from './src-file-extension-mapping';
 
-export async function command(commandOptions: CommandOptions) {
-  const {cwd, config} = commandOptions;
-
-  // Start with a fresh install of your dependencies, for production
-  const installCommandOptions = merge(commandOptions, {
-    config: {
-      installOptions: {
-        dest: BUILD_DEPENDENCIES_DIR,
-        env: {NODE_ENV: process.env.NODE_ENV || 'production'},
-        treeshake: config.installOptions.treeshake ?? true,
-      },
+async function installOptimizedDependencies(
+  allBuiltJsFiles: [string, string][],
+  installDest: string,
+  commandOptions: CommandOptions,
+) {
+  console.log('optimizing dependencies...');
+  const installConfig = merge(commandOptions.config, {
+    installOptions: {
+      dest: installDest,
+      env: {NODE_ENV: process.env.NODE_ENV || 'production'},
+      treeshake: commandOptions.config.installOptions.treeshake ?? true,
     },
   });
-  const dependencyImportMapLoc = path.join(BUILD_DEPENDENCIES_DIR, 'import-map.json');
-
-  // Start with a fresh install of your dependencies, always.
-  console.log(chalk.yellow('! rebuilding dependencies...'));
-  await installCommand(installCommandOptions);
+  // 1. Scan imports from your final built JS files.
+  const installTargets = await getInstallTargets(installConfig, allBuiltJsFiles);
+  // 2. Install dependencies, based on the scan of your final build.
+  const installResult = await installRunner(
+    {...commandOptions, config: installConfig},
+    installTargets,
+  );
+  // 3. Print stats immediate after install output.
+  if (installResult.stats) {
+    console.log(printStats(installResult.stats));
```

Comment:
Could this `printStats()` live in the install runner? Or are there times when we don’t want the install command to write to stdout?

Reply:
Yup, this is in preparation for some of the work happening in https://github.com/pikapkg/snowpack/pull/523, and a time when we'll be able to configure `install()` to not write to stdout/stderr, if you want to call it programatically.